federation/pkg/federation-controller/namespace/namespace_controller.go

```diff
@@ -371,25 +413,21 @@ func (nc *NamespaceController) delete(namespace *api_v1.Namespace) error {
 		return fmt.Errorf("failed to delete events list from namespace: %v", err)
 	}
 
-	// Remove kube_api.FinalzerKubernetes
+	// Delete the namespace from all underlying clusters.
+	updatedNamespaceObj, err := nc.deletionHelper.ProcessDeletion(updatedNamespace)
+	if err != nil {
+		return err
+	}
+
+	// Remove kube_api.FinalizerKubernetes
```

Comment:
Where is it blocked? It seems the [UpdateOnError](https://github.com/kubernetes/kubernetes/pull/34648/files#diff-2c30b9c3878f0bf2ea184f5d3b9c30c3R99) is not blocking.